src/copy.c

```diff
@@ -388,10 +388,34 @@ copyfrom(CopyChunkState *ccstate, List *range_table, Hypertable *ht, void (*call
 				ExecConstraints(check_resultRelInfo, myslot, estate);
 			}
 
-			if (cis->compress_state)
+			if (cis->compress_info)
 			{
 				TupleTableSlot *compress_slot =
-					ts_cm_functions->compress_row_exec(cis->compress_state, myslot);
+					ts_cm_functions->compress_row_exec(cis->compress_info->compress_state, myslot);
+				/* After Row triggers do not work with compressed chunks. So
+				 * explicitly call cagg trigger here
+				 */
+				if (cis->compress_info->has_cagg_trigger)
+				{
+					Assert(ts_cm_functions->continuous_agg_call_invalidation_trigger);
+					HeapTupleTableSlot *hslot = (HeapTupleTableSlot *) myslot;
+					if (!hslot->tuple)
+						hslot->tuple = heap_form_tuple(myslot->tts_tupleDescriptor,
+													   myslot->tts_values,
+													   myslot->tts_isnull);
+
+					ts_cm_functions->continuous_agg_call_invalidation_trigger(
+						ht->fd.id,
+						cis->rel,
+						hslot->tuple,
+						NULL /* chunk_newtuple */,
+						false /* update */,
+						false /* is_distributed_hypertable
+							   */
+						,
```

Comment:
That's strange formatting :) Maybe it would be a bit better to use a couple of const bool's instead?